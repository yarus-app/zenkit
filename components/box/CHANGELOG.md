# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.3.2](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.3.1...@zenkit/box@2.3.2) (2020-12-05)

**Note:** Version bump only for package @zenkit/box

## [2.3.1](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.3.0...@zenkit/box@2.3.1) (2020-11-26)

**Note:** Version bump only for package @zenkit/box

# [2.3.0](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.2.1...@zenkit/box@2.3.0) (2020-11-26)

### Features

-   **box:** added relative position ([f6d291c](https://github.com/yarus-app/zenkit/commit/f6d291cc177aa4776e9939471ddc57f56bd73b49))

## [2.2.1](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.2.0...@zenkit/box@2.2.1) (2020-11-26)

### Bug Fixes

-   **box:** fix box prop inline ([6aa2d53](https://github.com/yarus-app/zenkit/commit/6aa2d53f1429cfd7024a3ea83f234bcb9afe0811))

# [2.2.0](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.1.0...@zenkit/box@2.2.0) (2020-11-26)

### Features

-   **box:** add box prop inline ([766da48](https://github.com/yarus-app/zenkit/commit/766da48d1cadba875a04d65b76945e5a4ef51043))

# [2.1.0](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.0.3...@zenkit/box@2.1.0) (2020-11-26)

### Features

-   **box:** change box styles ([9a2a6c6](https://github.com/yarus-app/zenkit/commit/9a2a6c68cad4315313f95cc1ca13dd6a1f21ca2b))

## [2.0.3](https://github.com/yarus-app/zenkit/compare/@zenkit/box@2.0.2...@zenkit/box@2.0.3) (2020-11-26)

**Note:** Version bump only for package @zenkit/box

## 2.0.2 (2020-11-26)

### Bug Fixes

-   lerna version message ([58a6283](https://github.com/yarus-app/zenkit/commit/58a628369a47a9f7364456f6fa71aa6ce84fe958))

## 2.0.1 (2020-11-26)

**Note:** Version bump only for package @zenkit/box
