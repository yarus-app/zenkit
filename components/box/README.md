# [ZenKit](https://github.com/yarus-app/zenkit) / [Box](https://github.com/yarus-app/zenkit/tree/main/components/box)

---

Made with ❤️ by [Yaroslav Usenko](https://github.com/yar-usenko).

It is free software, and may be redistributed under the `MIT License` detailed in the [LICENSE.md](https://github.com/yarus-app/zenkit/blob/main/LICENSE.md) file.
