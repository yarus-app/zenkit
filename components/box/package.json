{
  "name": "@zenkit/box",
  "version": "2.2.1",
  "description": "",
  "keywords": [],
  "homepage": "https://github.com/yarus-app/zenkit/tree/main/components/box",
  "bugs": "https://github.com/yarus-app/zenkit/labels/component%3Abox",
  "license": "MIT",
  "author": "Yaroslav Usenko <usenko.jaroslav@gmail.com> (https://usenko.pp.ua)",
  "main": "lib/zenkit-box.js",
  "module": "lib/zenkit-box.mjs",
  "modules.root": "lib",
  "files": [
    "lib"
  ],
  "devDependencies": {
    "react": "16",
    "react-dom": "16",
    "styled-components": "^5.2.1"
  },
  "peerDependencies": {
    "react": "16",
    "react-dom": "16",
    "styled-components": "^5.2.1"
  },
  "publicConfig": {
    "access": "public"
  }
}
