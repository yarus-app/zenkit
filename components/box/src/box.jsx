import styled from 'styled-components/macro';

export const Box = styled.div`
  position: relative;
  display: ${({ inline }) => (inline ? 'inline-block' : 'block')};
  box-sizing: border-box;
`;

export default Box;
