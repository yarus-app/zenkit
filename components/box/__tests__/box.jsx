/* eslint-disable import/no-unresolved */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';
import { Box } from '@zenkit/box';

describe('<Box />', () => {
  test('should render correctly in "debug" mode', () => {
    const component = shallow(<Box />);

    expect(component).toMatchSnapshot();
  });
});
