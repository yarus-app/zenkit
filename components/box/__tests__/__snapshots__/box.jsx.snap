// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render correctly in "debug" mode 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  color: red;
}

<div
  className="c0"
/>
`;
